js/pokemon.js

```diff
@@ -1146,3 +1208,30 @@ function loadState() {
         }
     }
 }
+
+Array.prototype.equals = function (array) {
```

Comment:
Underscore.js has comparison functions -- prototype extensions are nice but in a wider project maintaining them with respect to multiple libraries is annoying rather than difficult.